x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ilm/AllocationRoutedStepTests.java

```diff
@@ -267,11 +258,20 @@ public void testExecuteAllocateNotCompleteOnlyOneCopyAllocated() throws Exceptio
                 .addShard(TestShardRouting.newShardRouting(new ShardId(index, 0), "node2", primaryOnNode1 == false,
                         ShardRoutingState.STARTED));
 
+
         AllocationRoutedStep step = new AllocationRoutedStep(randomStepKey(), randomStepKey());
+        logger.info("running test with routing configurations:\n\t includes: [{}]\n\t excludes: [{}]\n\t requires: [{}]",
+            mapToString(includes), mapToString(excludes), mapToString(requires));
         assertAllocateStatus(index, 2, 0, step, existingSettings, node1Settings, node2Settings, indexRoutingTable,
                 new ClusterStateWaitStep.Result(false, allShardsActiveAllocationInfo(0, 1)));
     }
 
+    private static String mapToString(Map<String, String> map) {
```

Comment:
We shouldn't need this function, `Map.toString()` already returns things in a format like `{key=value, food=eggplant}`

Reply:
This would be the poster for "software developer over-engineers task, and feels good about it too" :) Thanks for spotting it